internal/configs/transportserver.go

```diff
@@ -112,7 +116,55 @@ func generateStreamUpstreams(transportServerEx *TransportServerEx, upstreamNamer
 		upstreams = append(upstreams, ups)
 	}
 
-	return upstreams
+	return upstreams, hc
+}
+
+func generateTransportServerHealthCheck(healthCheck *conf_v1alpha1.HealthCheck) *version2.StreamHealthCheck {
+	if healthCheck == nil || !healthCheck.Enabled {
+		return nil
+	}
+
+	hc := generateTransportServerHealthCheckWithDefaults()
+
+	hc.Enabled = healthCheck.Enabled
+
+	if healthCheck.Interval != "" {
```

Comment:
sorry, didn't catch it during the previous round.

for time fields - interval, jitter and timeout --  we need to use `generateTime`, as that function also handles transformation cases like this  "10m 1s" -> "10m1s" , which prevents NGINX from failing to reload